plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -56,12 +56,23 @@
         extends BaseConnectorSmokeTest
 {
     protected final FileFormat format;
-    private final TrinoFileSystem fileSystem;
+    protected TrinoFileSystem fileSystem;
 
-    public BaseIcebergConnectorSmokeTest(FileFormat format)
+    protected BaseIcebergConnectorSmokeTest(FileFormat format)
     {
         this.format = requireNonNull(format, "format is null");
-        this.fileSystem = new HdfsFileSystemFactory(HDFS_ENVIRONMENT, HDFS_FILE_SYSTEM_STATS).create(TestingConnectorSession.SESSION);
+    }
+
+    protected abstract DistributedQueryRunner createIcebergQueryRunner()
+            throws Exception;
+
+    @Override
+    protected final QueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = createIcebergQueryRunner();
+        fileSystem = getFileSystemFactory(queryRunner).create(SESSION);
+        return queryRunner;
```

Comment:
make it `@BeforeClass` configuration method.

Reply:
This is implementing the abstract method from `AbstractTestQueryFramework`.

Reply:
Thanks, that's much cleaner. I confirmed that TestNG guarantees to run `@BeforeClass` in the superclass first.

>  In that case, TestNG guarantees that the "@Before" methods are executed in inheritance order (highest superclass first, then going down the inheritance chain), and the "@After" methods in reverse order (going up the inheritance chain).